assets/js/src/newsletters/sending_status.jsx

```diff
@@ -0,0 +1,185 @@
+import React from 'react';
+import MailPoet from 'mailpoet';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { Link } from 'react-router-dom';
+import Listing from 'listing/listing.jsx';
+import { CronMixin, MailerMixin } from 'newsletters/listings/mixins.jsx';
+
+const columns = [
+  {
+    name: 'subscriber_id',
+    label: MailPoet.I18n.t('subscriber'),
+    sortable: true,
+  },
+  {
+    name: 'status',
+    label: MailPoet.I18n.t('sendingStatus'),
+  },
+  {
+    name: 'failureReason',
+    label: MailPoet.I18n.t('failureReason'),
+  },
+];
+
+const messages = {
+  onNoItemsFound: () => MailPoet.I18n.t('noSendingTaskFound'),
+};
+
+const SendingStatus = (props) => {
+  const newsletterId = props.match.params.id;
+  const [isLoading, setIsLoading] = React.useState(true);
+  const [newsletterSubject, setNewsletterSubject] = React.useState('');
+
+  React.useEffect(() => {
+    MailPoet.Ajax.post({
+      api_version: window.mailpoet_api_version,
+      endpoint: 'newsletters',
+      action: 'get',
+      data: {
+        id: newsletterId,
+      },
+    })
+      .done((res) => {
+        setNewsletterSubject(res.data.subject);
+        setIsLoading(false);
+      })
+      .fail((res) => {
+        MailPoet.Notice.showApiErrorNotice(res);
+      });
+  }, [newsletterId]);
+
+  return (
+    <>
+      <h1>{MailPoet.I18n.t('sendingStatusTitle')}</h1>
+      <StatsLink
+        newsletterId={newsletterId}
+        newsletterSubject={newsletterSubject}
+      />
+      {!isLoading && (
```

Comment:
I understand what you did here, but the User Experience is quite bad now. User will come to a blank page, sees nothing and after a while, everything just jumps to the page. Before this change, there was a table showing them we're loading the data.

Reply:
I fixed that using `React.memo`